### PR TITLE
WP-329: Disallow non-localhost origins to perform autoenrolment

### PR DESCRIPTION
**Pull request by jplyle**
http://jira.webinos.org/browse/WP-329

The PZP will now check for websocket connections to make sure:

(1) They come from localhost
(2) That if they attempt to do autoenrolment, their origin hostname
is 'localhost' or '127.0.0.1'.

This is a quick fix to WP-329.

It also contains a fix to WP-341 (restricting websockets to localhost only)
http://jira.webinos.org/browse/WP-341

### CODE REVIEW

**Comment by jplyle** [comment]
I've emailed for help testing this pull request.  Might be worth leaving for a while.

**Comment by aldafu** [comment]
Sorry I didn't test this, just wanted to let you know your identation is off.

**Comment by jplyle** [comment]
@aldafu Could you be more specific?  Where?

The indentation for most of that file (and most other files) is off, in general.  Do we have an indentation "company style" anywhere?

**Comment by aldafu** [comment]
I wish we had a global style. Since we don't I think it's good practise to follow the style of the file you are editing. And Habib indents with two space characters.

File: webinos/pzp/lib/pzp_websocket.js [hunk 1]
```diff
@@ -283,6 +292,11 @@ var PzpWSS = function() {
     req.end();
   }
 
+ function approveRequest(request) {
+    var requestor = request.host.split(":")[0]; // don't care about port.
```
Comment (aldafu) [inline webinos/pzp/lib/pzp_websocket.js]: three spaces indent

File: webinos/pzp/lib/pzp_websocket.js [hunk 2]
```diff
@@ -308,15 +322,31 @@ var PzpWSS = function() {
...
             });
-
-            wsServer.on("connect", function(connection) {
-              logger.log("connection accepted.");
-              connectedApp(connection);
-              connection.on("message", function(message) { wsMessage(connection, message.utf8Data); });
-              connection.on("close", function(reason, description) { wsClose(connection, description) });
+            
+            wsServer.on("request", function(request) {
+                logger.log("Request for a websocket, origin: " + request.origin
```
Comment (aldafu) [inline webinos/pzp/lib/pzp_websocket.js]: four spaces indent

File: webinos/pzp/lib/pzp_websocket.js [hunk 3]
```diff
@@ -329,6 +359,8 @@ var PzpWSS = function() {
     });
   };
 
+
```
Comment (aldafu) [inline webinos/pzp/lib/pzp_websocket.js]: trailing whitespace

File: webinos/pzp/lib/pzp_websocket.js [hunk 4]
```diff
@@ -308,15 +322,31 @@ var PzpWSS = function() {
...
+                    });
+                    connection.on("close", function(reason, description) { 
+                        wsClose(connection, description) 
+                    });
+                } else {
+                    logger.error("Failed to accept websocket connection: " +
+                        "wrong host or origin");
+                }
             });
+
```
Comment (aldafu) [inline webinos/pzp/lib/pzp_websocket.js]: trailing whitespace

**Comment by jplyle** [comment]
I have used http://jsbeautifier.org/ to make the whole of pzp_websocket.js consistent.

**Comment by habibvirji** [comment]
Please add last line in commit msg of from Jira Issue: WP-329

**Comment by jplyle** [comment]
Only if you don't mind me adding a 'merge' to this commit.  I'm doing the following:
- git commit --amend 
- git pull origin WP-329-...
- git push origin WP-329-...

Should I be doing something else?

**Comment by habibvirji** [comment]
That should do this time.

**Pull request merged**